opslevel/import_utils.go

```diff
@@ -0,0 +1,59 @@
+package opslevel
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+func extractContactFromContacts(contactId opslevel.ID, contacts []opslevel.Contact) *opslevel.Contact {
+	for _, readContact := range contacts {
+		if contactId == readContact.Id {
+			return &readContact
+		}
+	}
+	return nil
+}
+
+func extractTagFromTags(tagId opslevel.ID, tags []opslevel.Tag) *opslevel.Tag {
+	for _, readTag := range tags {
+		if tagId == readTag.Id {
+			return &readTag
+		}
+	}
+	return nil
+}
+
+func extractToolFromTools(toolId opslevel.ID, tools []opslevel.Tool) *opslevel.Tool {
+	for _, readTool := range tools {
+		if toolId == readTool.Id {
+			return &readTool
+		}
+	}
+	return nil
+}
+
+func getTagsFromResource(client *opslevel.Client, resource opslevel.TaggableResourceInterface) (*opslevel.TagConnection, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	tags, err := resource.GetTags(client, nil)
+	if err != nil {
+		diags.AddError(
+			"opslevel client error",
+			fmt.Sprintf("unable to read tags on %s with id '%s', got error: %s", string(resource.ResourceType()), string(resource.ResourceId()), err),
```

Comment:
IMO this function does not need to return `diag.Diagnostics` and should return an `error` (or `nil`). This message wrapping the error is unnecessary and the message wrapping the error should be added to whatever is calling `getTagsFromResource` and checking the returned `diag.Diagnostics`. 

Also I think if we got the behaviour of `GetTags` on the interface to be consistent we would not need a function like this. I think this is better solved in opslevel-go by either fixing the implementations in place or creating a new implementation and deprecating the old one.

Reply:
The error needs to be wrapped here or from the calling function 🤷‍♂️ functionally there's little difference.

`GetTags` itself is a workaround for a fundamental shortcoming in opslevel-go. These util functions have limited scope and suffice for now. There are much bigger fish to fry 🐟